src/seedsigner/views/seed_views.py

```diff
@@ -323,20 +323,52 @@ def __init__(self):
 
 
     def run(self):
-        ret = self.run_screen(seed_screens.SeedAddPassphraseScreen, passphrase=self.seed.passphrase)
+        ret_passphrase, ret_btn = self.run_screen(seed_screens.SeedAddPassphraseScreen, passphrase=self.seed.passphrase)
 
-        if ret == RET_CODE__BACK_BUTTON:
-            return Destination(BackStackView)
-        
         # The new passphrase will be the return value; it might be empty.
-        self.seed.set_passphrase(ret)
-        if len(self.seed.passphrase) > 0:
+        self.seed.set_passphrase(ret_passphrase)
+        if ret_btn == RET_CODE__BACK_BUTTON:
```

Comment:
Minor nit: I try to keep different logical code sections separated by whitespace. So I'd normally separate lines 329 and 330 here.